### ISSUE TITLE
MM 2.3.18 hangs after click on link to local .txt file in the Previewer

### ISSUE THREAD

**Issue by jbridgy** [RickStrahl/MarkdownMonster#922]
Clicking into the Preview of MM 2.3.18 on a link to a local text file makes MM completely inoperable. You have to terminate MM by the Task Manager.

This happens regardless whether extension .txt is associated with Notepad (default) or another editor.
The same is true for *.log and *.ini files.

Example:
```markdown
[Plain.txt (abs. path)](C:/mm_test_link_to_txt/Plain.txt)

[Plain.txt (rel. path)](./Plain.txt)
```

Plain.txt just contains:
```plaintext
Greetings from plain text file.
```

**Comment by RickStrahl** [comment]
Not sure what you're describing. Click where exactly? In the preview? Or the Editor?

Either way, when I click (or ctrl-click) on either of those nothing happens because it's not a URL. It's supposed to open a browser.

![FileLinkClicks](https://user-images.githubusercontent.com/1374013/157541060-ac2c6951-9dfc-4528-a5ad-5f183a265b9e.gif)

**Comment by jbridgy** [comment]
Strange!? On my machine the link to Plain.md which is a copy of Plain.txt works but the link to Plain.txt does not (as already reported).

![link_to_txt_fails](https://user-images.githubusercontent.com/98041047/157720205-4f7818e7-8d7e-4f84-b864-9668a31b6a79.gif)

**Comment by RickStrahl** [comment]
Ok looking at this again tonight - I see the hang. the hang is from the previewer, but not from the editor.

**Comment by RickStrahl** [comment]
Fixed for 2.4.5.2.

This isssue is caused by the lack of proper async support in the WebView2 control, so MM now has to call synchronous methods change down to sync operation. Previously did this without explicit 

Original code:

```cs
public bool PreviewLinkNavigation(string url, string src = null)
{
    // HACK: We shouldn't use the sync version but WebView fails with async call
    //       so this is required for now. Check in future version (2.0.0.1)
    var editor = Model.ActiveEditor;
    
    // this goes badly and locks up
    bool result = editor.PreviewLinkNavigation(url, src).GetAwaiter().GetResult();
    return result;
}
```

This works (but still has potential for lockup):

```cs
public bool PreviewLinkNavigation(string url, string src = null)
{
    // HACK: We shouldn't use the sync version but WebView fails with async call
    //       so this is required for now. Check in future version (2.0.0.1)
    var editor = Model.ActiveEditor;
    bool result = Model.Window.Dispatcher.RunSync<bool>(() => editor.PreviewLinkNavigation(url, src));
    return result;
}
```

This can still lock up, but given this is a low impact request that is 'one and done' it's unlikely to run into a race condition that locks it up now, so in the short term this is the fix.

Ultimately this and a handful of other WebView2 interaction operations have to be made `async` when the WebView2 control starts properly supporting async calls from the browser.

Related issue on the WebView Feedback site:

https://github.com/MicrosoftEdge/WebView2Feedback/issues/822

**Comment by RickStrahl** [comment]
Playing around with this some more I see this can still hang up MM in some cases where the WebBrowser controls (both editor and preview) become unresponsive. Apparently this happens when trying to open a file that doesn't exist.

For now I'm going to leave this one as is as according to Microsoft the async issue will be addressed soon. At that point the code can use proper async semantics which would fix this mess properly. ANything else that I do here in the current state will not be 100% due to the sync->async transition.

Leaving this open for a few weeks to see if I can revisit once the WebView has the async updates that have been promised soon.
